base_layer/core/src/transactions/transaction_components/encrypted_data.rs

```diff
@@ -114,28 +121,33 @@ impl EncryptedData {
         encrypted_data: &EncryptedData,
     ) -> Result<(MicroTari, PrivateKey), EncryptedDataError> {
         // Extract the nonce and ciphertext
-        let binding = encrypted_data.to_byte_vec();
-        let (nonce, ciphertext) = binding.split_at(size_of::<XNonce>());
+        let (nonce, ciphertext) = encrypted_data.as_bytes().split_at(size_of::<XNonce>());
         let nonce_ga = XNonce::from_slice(nonce);
 
+        // Set up a buffer for decryption
+        let mut buffer = HeaplessVec::<u8, { size_of::<u64>() + PrivateKey::KEY_LEN + size_of::<Tag>() }>::new();
+        buffer.extend_from_slice(ciphertext).map_err(|_| EncryptedDataError::BufferError)?;
+    
+        // Set up the AEAD
         let aead_key = kdf_aead(encryption_key, commitment);
         let cipher = XChaCha20Poly1305::new(GenericArray::from_slice(aead_key.reveal()));
-        let aead_payload = Payload {
-            msg: ciphertext,
-            aad: Self::TAG,
-        };
-        let decrypted_bytes = cipher.decrypt(nonce_ga, aead_payload)?;
-        let mut value_bytes = [0u8; VALUE_SIZE];
-        value_bytes.clone_from_slice(&decrypted_bytes[0..VALUE_SIZE]);
-        let mut mask_bytes = [0u8; KEY_SIZE];
-        mask_bytes.clone_from_slice(&decrypted_bytes[VALUE_SIZE..VALUE_SIZE + KEY_SIZE]);
+
+        // Decrypt in place using the buffer
+        cipher.decrypt_in_place(nonce_ga, ENCRYPTED_DATA_AAD, &mut buffer)?;
```

Comment:
Maybe worthwhile adding this
```suggestion
        cipher.decrypt_in_place(nonce_ga, ENCRYPTED_DATA_AAD, &mut buffer)
            .map_err(|e| {
                buffer.zeroize();
                EncryptedDataError::DecryptionFailed(e)
            })?;
```

Reply:
I also could not test `fn esmeralda_genesis_sanity_check()` and `fn igor_genesis_sanity_check()` due to `error: failed to get `tari_crypto` as a dependency of package `tari_common v0.50.0-pre.2`, but I assume it will pass - the manual manipulation of the json file looks correct

Reply:
> Maybe worthwhile adding this

I assumed that failed decryption would either result in no plaintext being present in the buffer, or would be caused by manipulated ciphertext, both of which would mean zeroizing isn't very helpful. But you're right that it would be worthwhile just in case, even if it's a minor performance hit. Will update.

Reply:
> I also could not test

Not sure why this didn't work for you, but I suppose we can rely on CI for this?

Reply:
> Not sure why this didn't work for you

Looks like this was caused by my foolish deletion of the referenced branch after the corresponding PR was merged. Will update accordingly.

Reply:
> Maybe worthwhile adding this

Working on a change that uses an array instead, which should render the closure moot.